test/sql-luatest/view_test.lua

```diff
@@ -0,0 +1,437 @@
+        t.assert_error_msg_equals(exp_err, box.space._space.replace,
+                                  box.space._space, view)
+
+        --
+        -- Try to change engine.
+        --
+        view = box.space._space.index[2]:select('v')[1]:totable()
+        view[4] = 'a'
+        exp_err = "Can't modify space 'v': view can not be altered"
+        t.assert_error_msg_equals(exp_err, box.space._space.replace,
+                                  box.space._space, view)
+
+        --
+        -- Try to add a field.
+        --
+        view = box.space._space.index[2]:select('v')[1]:totable()
+        local view_format = box.space.v:format()
+        local f = {
+            type = 'string',
+            nullable_action = 'none',
+            name = 'B',
+            is_nullable = true,
+        }
+        table.insert(view_format, f)
+        view[5] = 2
+        view[7] = view_format
+        exp_err = "Can't modify space 'v': view can not be altered"
+        t.assert_error_msg_equals(exp_err, box.space._space.replace,
+                                  box.space._space, view)
+
+        --
+        -- Try to modify format only.
+        --
+        view = box.space.v
+        exp_err = "Can't modify space 'v': view can not be altered"
+        t.assert_error_msg_equals(exp_err, view.format, view, {})
+
+        view_format = box.space.v:format()
+        view_format[1].name = 'B'
+        exp_err = "Can't modify space 'v': view can not be altered"
+        t.assert_error_msg_equals(exp_err, view.format, view, view_format)
+
+        box.execute("DROP VIEW v;")
+        box.execute("DROP TABLE t1;")
+    end)
+end
+
+--
+-- Make sure, that it is possible to create a VIEW which
+-- refers to "_v" space, i.e. to sysview engine.
+--
+g.test_4111_format_in_sysview = function(cg)
+    cg.server:exec(function(engine)
+
+        t.assert(box.space._vspace.index[1]:count(1) > 0)
+
+        box.execute([[CREATE VIEW v AS SELECT name, engine FROM _vspace;]])
+        t.assert(box.space.v ~= nil)
+        local exp = {
+            metadata = {
+                {name = "name", type = "string"},
+                {name = "engine", type = "string"},
+            },
+            rows = {{'v', engine}},
+        }
+        local res = box.execute([[SELECT * FROM v WHERE name = 'v';]])
+        t.assert_equals(res, exp)
+        box.execute([[DROP VIEW v;]])
+    end, {cg.params.engine})
+end
+
+--
+-- Make sure, that user can't access to protected tables via VIEW
+--
+g.test_4104_view_access_check = function(cg)
+    cg.server:exec(function()
+        box.execute("CREATE TABLE supersecret(id INT PRIMARY KEY, data TEXT);")
+        t.assert(box.space.supersecret ~= nil)
+        box.execute("CREATE TABLE supersecret2(id INT PRIMARY KEY, data TEXT);")
+        t.assert(box.space.supersecret2 ~= nil)
+        box.execute([[INSERT INTO supersecret
+                             VALUES(1, 'very very big secret');]])
```

Comment:
There seems to be something wrong with the indentation here and below. I think it should be like this:
```Lua
        box.execute([[INSERT INTO supersecret
                      VALUES(1, 'very very big secret');]])
```

Also, here you can use this:
```Lua
        box.execute([[INSERT INTO supersecret VALUES(1, 'very big secret');]])
```